codegen/test/chart-no-desc/templates/deployment.yaml

```diff
@@ -100,7 +111,17 @@ spec:
             cpu: 500m
             memory: 256Mi
 {{- end}}
+        {{/*
+          Render securityContext configs if it is set or an empty map.
+          Otherwise if it is false render an empty map. If securityContext is not set render default securityContext.
+        */}}
         securityContext:
+{{- if or ($palette.securityContext) (eq "map[]" (printf "%v" $palette.securityContext)) }}
+{{ toYaml $palette.securityContext | indent 10}}
+{{/* Because securityContext is nil by default we can only perform following conversion if it is a boolean. Skip conditional otherwise. */}}
+{{- else if eq (ternary $palette.securityContext true (eq "bool" (printf "%T" $palette.securityContext))) false }}
```

Comment:
this got more complicated and I don't love it but I needed to add this as the GME helm tests were running into the following error:

```
<string>: Error: template: gloo-mesh-enterprise/templates/deployment.yaml:111:3: executing "gloo-mesh-enterprise/templates/deployment.yaml" at <include "skv2.utils.merge" (list . $glooMeshMgmtServerDeploymentOverrides "gloo-mesh-mgmt-server.deploymentSpec")>: error calling include: template: gloo-mesh-enterprise/templates/_helpers.tpl:47:22: executing "skv2.utils.merge" at <include (index . 2) $top>: error calling include: template: gloo-mesh-enterprise/templates/deployment.yaml:85:12: executing "gloo-mesh-mgmt-server.deploymentSpec" at <eq false $glooMeshMgmtServer.securityContext>: error calling eq: invalid type for comparison```